src/org/zaproxy/zap/extension/stats/ExtensionStats.java

```diff
@@ -0,0 +1,163 @@
+	private StatsParam statsParam = null;
+
+    private Logger logger = Logger.getLogger(ExtensionStats.class);
+
+	public ExtensionStats() {
+		super();
+		initialize();
+	}
+
+	private void initialize() {
+        this.setName(NAME);
+	}
+
+	@Override
+	public void hook(ExtensionHook extensionHook) {
+	    super.hook(extensionHook);
+
+		extensionHook.addOptionsChangedListener(this);
+		extensionHook.addOptionsParamSet(getStatsParam());
+
+	    if (getView() != null) {
+	        extensionHook.getHookView().addOptionPanel(getOptionsStatsPanel());
+	    }
+	    
+	    API.getInstance().registerApiImplementor(statsAPI);
+	    statsAPI.addApiOptions(getStatsParam());
+
+	}
+
+	@Override
+	public void optionsLoaded() {
+		optionsChanged();
+	}
+
+	private OptionsStatsPanel getOptionsStatsPanel() {
+		if (optionsStatsPanel == null) {
+			optionsStatsPanel = new OptionsStatsPanel(this);
+		}
+		return optionsStatsPanel;
+	}
+	
+	private StatsParam getStatsParam() {
+		if (statsParam == null) {
+			statsParam = new StatsParam();
+		}
+		return statsParam;
+	}
+	
+	@Override
+	public boolean isCore() {
+		return true;
+	}
+	
+	@Override
+	public String getAuthor() {
+		return Constant.ZAP_TEAM;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("stats.desc");
+	}
+
+	@Override
+	public URL getURL() {
+		try {
+			return new URL(Constant.ZAP_HOMEPAGE);
+		} catch (MalformedURLException e) {
+			return null;
+		}
+	}
+
+	/**
+	 * No database tables used, so all supported
+	 */
+	@Override
+	public boolean supportsDb(String type) {
+    	return true;
+    }
+
+	@Override
+	public void optionsChanged(OptionsParam optionsParam) {
+		optionsChanged();
+	}
+	
+	private void optionsChanged() {
+        if ((inMemStats == null) == this.getStatsParam().isInMemoryEnabled()) {
+        	// Somethings changed
+        	if (inMemStats == null) {
+            	logger.info("Start recording in memory stats");
+            	inMemStats = new InMemoryStats();
+            	Stats.addListener(inMemStats);
+        	} else {
+            	logger.info("Stop recording in memory stats");
+            	Stats.removeListener(inMemStats);
+            	inMemStats = null;
+        	}
+        }
+        if ((statsd == null) == this.getStatsParam().isStatsdEnabled()) {
+        	// Somethings changed
+        	if (statsd == null) {
+            	logger.info("Start sending stats to statsd server");
+            	try {
+    				statsd = new Statsd(
+    							this.getStatsParam().getStatsdHost(),
+    							this.getStatsParam().getStatsdPort(),
+    							this.getStatsParam().getStatsdPrefix());
+    	        	Stats.addListener(statsd);
+    			} catch (Exception e) {
+    				logger.error(e.getMessage(), e);
+    			}
+        	} else {
+            	logger.info("Stop sending stats to statsd server");
+            	Stats.removeListener(statsd);
+            	statsd = null;
+        	}
+        } else if (statsd != null) {
```

Comment:
Changes to the port are not applied if the `statsd` client was already initialised. Is that expected?